pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -479,6 +479,13 @@ public void sendAsync(Message<?> message, SendCallback callback) {
                 }
                 String uuid = totalChunks > 1 ? String.format("%s-%d", producerName, sequenceId) : null;
                 for (int chunkId = 0; chunkId < totalChunks; chunkId++) {
+                    // Need to reset the schemaVersion, because the schemaVersion is a ByteBuf object in
+                    // `MessageMetadata`, if we want to re-serialize the `SEND` command using a same `MessageMetadata`,
+                    // we need to reset the ByteBuf of the schemaVersion in `MessageMetadata`, I think we need to
+                    // reset `ByteBuf` objects in `MessageMetadata` after call the method `MessageMetadata#writeTo()`.
+                    if (chunkId > 0 && msg.getMessageBuilder().hasSchemaVersion()) {
```

Comment:
seems getSchemaVersion() will copy data around, might be good if we just store the result as it won't change

Reply:
If we want to call the method `writeTo()` of the class `MessageMetadata` multi times, it's better to modify the logic of the `MessageMetadata`, the read index of `ByteBuf` object should be reset after writing.